impeller/compiler/reflector.cc

```diff
@@ -376,16 +376,16 @@ std::shared_ptr<RuntimeStageData> Reflector::GenerateRuntimeStageData() const {
   return data;
 }
 
-uint32_t Reflector::GetArrayElements(const spirv_cross::SPIRType& type) const {
+std::optional<uint32_t> Reflector::GetArrayElements(
+    const spirv_cross::SPIRType& type) const {
   if (type.array.empty()) {
-    return 0;
-  }
-  uint32_t elements = 1;
-  for (size_t i = 0; i < type.array.size(); i++) {
-    FML_CHECK(type.array_size_literal[i]);
-    elements *= type.array[i];
+    return std::nullopt;
   }
-  return elements;
+  FML_CHECK(type.array.size() == 1)
```

Comment:
Instead of making the compiler crash, perhaps just return nullopt (or propagate the error up the stack) with a validation log?

Reply:
Is there an example somewhere of how we're doing error propagation in the compiler?

I assumed crashing is the right thing to do here because compilation is failing and we won't be able to output valid reflection headers or valid SPIRV that can be used on GLES 2.0.

Reply:
We could maybe support non-literal sizes for the array but that wasn't implemented already and I figure that can wait until another patch.

Reply:
Elsewhere in the reflector, there is a pattern of returning optionals and checking their values before adding a validation log and tripping validity. `ReadStructMembers` just returns a vector ATM which should ideally be an optional vector with a check.